prover/vk_setup_data_generator_server_fri/src/vk_generator.rs

```diff
@@ -27,11 +56,125 @@ fn main() -> anyhow::Result<()> {
+            witness: AtomicCell::new(Some(
+                EcrecoverCircuitInstanceWitness::<GoldilocksField>::default(),
+            )),
+            config: Arc::new(geometry.cycles_per_ecrecover_circuit as usize),
+            round_function: Arc::new(Poseidon2Goldilocks),
+            expected_public_input: None,
+        }),
+        ZkSyncBaseLayerCircuit::RAMPermutation(ZkSyncUniformCircuitInstance {
+            witness: AtomicCell::new(Some(
+                RamPermutationCircuitInstanceWitness::<GoldilocksField>::default(),
+            )),
+            config: Arc::new(geometry.cycles_per_ram_permutation as usize),
+            round_function: Arc::new(Poseidon2Goldilocks),
+            expected_public_input: None,
+        }),
+        ZkSyncBaseLayerCircuit::StorageSorter(ZkSyncUniformCircuitInstance {
+            witness: AtomicCell::new(Some(
+                StorageDeduplicatorInstanceWitness::<GoldilocksField>::default(),
+            )),
+            config: Arc::new(geometry.cycles_per_storage_sorter as usize),
+            round_function: Arc::new(Poseidon2Goldilocks),
+            expected_public_input: None,
+        }),
+        ZkSyncBaseLayerCircuit::StorageApplication(ZkSyncUniformCircuitInstance {
+            witness: AtomicCell::new(Some(StorageApplicationCircuitInstanceWitness::<
+                GoldilocksField,
+            >::default())),
+            config: Arc::new(geometry.cycles_per_storage_application as usize),
+            round_function: Arc::new(Poseidon2Goldilocks),
+            expected_public_input: None,
+        }),
+        ZkSyncBaseLayerCircuit::EventsSorter(ZkSyncUniformCircuitInstance {
+            witness: AtomicCell::new(Some(
+                EventsDeduplicatorInstanceWitness::<GoldilocksField>::default(),
+            )),
+            config: Arc::new(geometry.cycles_per_events_or_l1_messages_sorter as usize),
+            round_function: Arc::new(Poseidon2Goldilocks),
+            expected_public_input: None,
+        }),
+        ZkSyncBaseLayerCircuit::L1MessagesSorter(ZkSyncUniformCircuitInstance {
+            witness: AtomicCell::new(Some(
+                EventsDeduplicatorInstanceWitness::<GoldilocksField>::default(),
+            )),
+            config: Arc::new(geometry.cycles_per_events_or_l1_messages_sorter as usize),
+            round_function: Arc::new(Poseidon2Goldilocks),
+            expected_public_input: None,
+        }),
+        ZkSyncBaseLayerCircuit::L1MessagesHasher(ZkSyncUniformCircuitInstance {
+            witness: AtomicCell::new(Some(
+                LinearHasherCircuitInstanceWitness::<GoldilocksField>::default(),
+            )),
+            config: Arc::new(geometry.limit_for_l1_messages_pudata_hasher as usize),
+            round_function: Arc::new(Poseidon2Goldilocks),
+            expected_public_input: None,
+        }),
+    ]
+}
+
 pub fn generate_basic_circuit_vks() -> anyhow::Result<()> {
```

Comment:
Is it useful with such signature? Maybe change is to accept `&mut dyn DataSource` ?

Reply:
moved changes to test_harness directly.